File: frontend/android/MyApplication/app/src/main/java/com/example/myapplication/util/TokenManager.kt
package com.example.myapplication.util

import android.content.Context
import android.content.SharedPreferences

object TokenManager {
    private const val PREF_NAME = "auth_pref"
    private const val KEY_TOKEN = "jwt_token"
    private const val KEY_USERNAME = "username"

    private fun getPrefs(context: Context): SharedPreferences {
        return context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
    }

    fun saveToken(context: Context, token: String, username: String) {
        getPrefs(context).edit()
            .putString(KEY_TOKEN, token)
            .putString(KEY_USERNAME, username)
            .apply()
    }

    fun getToken(context: Context): String? {
        return getPrefs(context).getString(KEY_TOKEN, null)
    }

    fun getUsername(context: Context): String? {
        return getPrefs(context).getString(KEY_USERNAME, null)
    }

    fun clear(context: Context) {
        getPrefs(context).edit().clear().apply()
    }
}

File: frontend/react_native/src/utils/tokenManager.js
// src/utils/tokenManager.js

import AsyncStorage from '@react-native-async-storage/async-storage';

export const TokenManager = {
  saveToken: async (token, username) => {
    try {
      await AsyncStorage.setItem('jwt_token', token);
      await AsyncStorage.setItem('username', username);
    } catch (e) {
      console.error('Error saving token', e);
    }
  },

  getToken: async () => {
    try {
      return await AsyncStorage.getItem('jwt_token');
    } catch (e) {
      console.error('Error getting token', e);
      return null;
    }
  },

  getUsername: async () => {
    try {
      return await AsyncStorage.getItem('username');
    } catch (e) {
      console.error('Error getting username', e);
      return null;
    }
  },

  clear: async () => {
    try {
      await AsyncStorage.clear();
    } catch (e) {
      console.error('Error clearing token', e);
    }
  },
};
